src/pip/_internal/operations/prepare.py

```diff
@@ -376,109 +376,105 @@ def _download_should_save(self):
             "Could not find or access download directory '{}'"
             .format(self.download_dir))
 
-    def prepare_linked_requirement(
-        self,
-        req,  # type: InstallRequirement
-        parallel_builds=False,  # type: bool
-    ):
-        # type: (...) -> AbstractDistribution
-        """Prepare a requirement that would be obtained from req.link
-        """
-        assert req.link
-        link = req.link
-
-        # TODO: Breakup into smaller functions
-        if link.scheme == 'file':
+    def _log_preparing_link(self, req, link):
```

Comment:
This doesn't need to pass link into this, since `link = req.link` is performed outside. We can move that assignment in here.